rc/extra/editorconfig.kak

```diff
@@ -1,35 +1,62 @@
-def editorconfig-load -docstring "Set indentation options according to editorconfig file" %{
+# http://editorconfig.org/#file-format-details
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+
+decl -hidden bool editorconfig_trim_trailing_whitespace false
+decl -hidden int  editorconfig_insert_final_newline     0
+
+def -hidden _editorconfig-bufwritepre-hook %{
+    hook buffer BufWritePre %val{buffile} -group editorconfig-hooks %{ %sh{
+        if [ ${kak_opt_editorconfig_trim_trailing_whitespace} = "true" ]; then
+            printf %s\\n "try %{ exec -draft %{ %s\h+$<ret>d } }"
+        fi
+        if [ ${kak_opt_editorconfig_insert_final_newline} -gt 0 ]; then
```

Comment:
Kakoune always guarantees the file ends with a newline, because Kakounes buffers are arrays of lines and lines always ends with an end of line char. This seems to be inserting additional newlines, which is not how I interpret the editorconfig spec, I believe that option is there to guarantee the file ends with `\n`, which is what Kakoune will do anyway, and which seems to be the correct thing with text files in a posix environment.